### DISCUSSION TITLE
How to disable multi-caret editing?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#281] [General]
It's really annoying. When I select the text, then I really want to type something to replace it. For example, there are multiple instances of `int` and I selected the whole text `int* foo`. I really want to replace it with `double bar`, but the editor will assume I want to edit `int`.

p/s: And it will get worse if the language server `clangd` is running. Everything will become a mess!

**Comment by SpartanJ** [comment]
Sorry but I don't really understand what you're describing. If you select the text, any replacemant is done over the whole selection, if the cursor does not have selection it will try to select the surrounding word. Multi-cursor is cancelled with the escape key. Clangd (or any LSP) provides suggestions over where the text should be replaced and ecode will honor it since it's much more precise than selecting the surrounding word, this is the expected behavior. I guess you ask if it's possible to disable any kind of text replacement with the auto-complete selection?
